src/frontend/screens/Audio/ExistingRecording.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {useEffect, useState} from 'react';
+import {Pressable, View, StyleSheet} from 'react-native';
+import {
+  HeaderBackButton,
+  HeaderBackButtonProps,
+} from '@react-navigation/elements';
+import {useNavigation} from '@react-navigation/native';
+import {useIntl, defineMessages} from 'react-intl';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+import {Playback} from '../../sharedComponents/Playback';
+import {
+  BottomSheetModal,
+  BottomSheetModalContent,
+  useBottomSheetModal,
+} from '../../sharedComponents/BottomSheetModal';
+import {CloseIcon, DeleteIcon} from '../../sharedComponents/icons';
+import {WHITE, BLACK} from '../../lib/styles';
+import ErrorIcon from '../../images/Error.svg';
+import Share from 'react-native-share';
+import * as FileSystem from 'expo-file-system';
+import {UIActivityIndicator} from 'react-native-indicators';
+
+const m = defineMessages({
+  deleteBottomSheetTitle: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetTitle',
+    defaultMessage: 'Delete?',
+  },
+  deleteBottomSheetDescription: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetDescription',
+    defaultMessage:
+      'Your Audio Recording will be permanently deleted. This cannot be undone.',
+  },
+  deleteBottomSheetPrimaryButtonText: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetPrimaryButtonText',
+    defaultMessage: 'Delete',
+  },
+  deleteBottomSheetSecondaryButtonText: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetSecondaryButtonText',
+    defaultMessage: 'Cancel',
+  },
+});
+
+interface ExistingRecordingProps {
+  uri: string;
+  onDelete: () => void;
+  isEditing: boolean;
+}
+
+export const ExistingRecording: React.FC<ExistingRecordingProps> = ({
+  uri,
+  onDelete,
+  isEditing,
+}) => {
+  const {formatMessage: t} = useIntl();
+  const navigation = useNavigation();
+  const {sheetRef, isOpen, openSheet, closeSheet} = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const [localUri, setLocalUri] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerShown: true,
+      headerLeft: (props: HeaderBackButtonProps) => (
+        <HeaderBackButton
+          {...props}
+          onPress={() => navigation.goBack()}
+          backImage={backImageProps => (
+            <CloseIcon color={backImageProps.tintColor} />
+          )}
+        />
+      ),
+    });
+  }, [navigation]);
+
+  const handleDelete = () => {
+    closeSheet();
+    onDelete();
+  };
+
+  const handleShare = async () => {
+    setLoading(true);
+    try {
+      if (!localUri) {
+        const tempFileName = `audio_${Date.now()}.m4a`;
+        const localFilePath = `${FileSystem.cacheDirectory}${tempFileName}`;
+        const downloadResult = await FileSystem.downloadAsync(
+          uri,
+          localFilePath,
+        );
```

Comment:
This is breaking when I try and share an audio file before it has been written to the db (aka when I am creating and observation and have not saved the observation yet.

This is the error I am getting:
`Error sharing file: [Error: Call to function 'ExponentFileSystem.downloadAsync' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but was 'file']`

Idk if we necessarily want to be sharing an audio file before it is saved? Unless design explicitly wanted that, I think we should follow the pattern of the rest of the observations and only allow them to share after it isaved

Reply:
Sounds good. I will remove file sharing if file hasn't been saved yet